src/org/zaproxy/zap/extension/ascanrules/TestCrossSiteScriptV2.java

```diff
@@ -153,7 +160,13 @@ public void scan(HttpMessage msg, String param, String value) {
             	// No luck - try again, appending the eyecatcher to the original value
     			msg2 = getNewMsg();
     			setParameter(msg2, param, value + Constant.getEyeCatcher());
-                sendAndReceive(msg2);
+                try {
+        			sendAndReceive(msg2);
+            	} catch (InvalidRedirectLocationException|UnknownHostException e) {
+            		// Not an error, just means we probably attacked the redirect location

```

Comment:
Same case here? If so, then the catch block can be left in the outer try statement.